ansible/roles/ocp4-workload-integreatly/tasks/workload.yml

```diff
@@ -33,6 +33,72 @@
   - "Variable 2:      {{ ocp4_workload_integreatly.variable_2 }}"
   - "Variable Secret: {{ ocp4_workload_integreatly.variable_secret }}"
 
+- name: Get cluster token
+  shell: oc whoami -t
+  register: oc_login_token
+
+- name: Get existing htpasswd
+  shell: oc get secret htpasswd-secret -n openshift-config -o json | jq -j '.data.htpasswd' | base64 --decode > /tmp/htpasswd
+
+- name: Check if htpasswd exists
+  stat: 
+    path: /tmp/htpasswd
+  register: result
+
+- name: Backup htpasswd if one exists
+  shell: cp /tmp/htpasswd /tmp/htpasswd.backup
+  when: result.stat.exists
+
+- name: Create htpasswd if one does not exist 
+  command: touch /tmp/htpasswd
+  when: result.stat.exists == false
+
+- name: Generate evals password
+  command: openssl rand -base64 12
+  register: openssl_output
+
+- name: Set evals password
+  set_fact:
+    evals_password: "{{ openssl_output.stdout }}"
+
+- name: Create evals users
+  shell: htpasswd -B -b /tmp/htpasswd {{ evals_user_name_format|format(item|int) }} {{ evals_password }}
+  with_sequence: count={{ user_count }}
+
+- name: Remove existing htpasswd secret
+  shell: oc delete secret htpasswd-secret -n openshift-config
+  when: result.stat.exists
+
+- name: Create new htpasswd secret
+  shell: oc create secret generic htpasswd-secret --from-file=htpasswd=/tmp/htpasswd -n openshift-config
+
+- name: Backup oauth/cluster from openshift-config namespace to tmp/cluster-oauth.backup.json
+  shell: oc get oauth/cluster -n openshift-config -o json > /tmp/cluster-oauth.backup.json
+
+- name: Patch cluster OAuth with htpasswd
+  shell: oc patch oauth/cluster --type json -n openshift-config -p '[{"op":"add","path":"/spec/identityProviders/-","value":{"mappingMethod":"claim","name":"htpasswd_provider","type":"HTPasswd","htpasswd":{"fileData":{"name":"htpass-secret"}}}}]'
+  when: result.stat.exists == false
+
+- name: Give cluster OAuth 30 seconds to adapt to changes
```

Comment:
Is there any resource we could poll instead? I'm always cautious of sleep statements

Reply:
One thing I'd recommend is to use `k8s`everywhere possible instead of oc shell commands.

Technically you could poll the cluster operator and make sure it's done rolling out.

Do you need to *do* anything as any of these users? If not it should be safe to just continue without waiting.

Reply:
I will remove the intialise eval users, we can then remove this wait as it will not be needed.